frontend/src/app/accountCreation/AccountCreationApp.tsx

```diff
@@ -22,67 +26,77 @@ import { MfaPhoneVerify } from "./MfaPhoneVerify/MfaPhoneVerify";
 import { MfaOktaVerify } from "./MfaOktaVerify/MfaOktaVerify";
 import { MfaGoogleAuthVerify } from "./MfaGoogleAuthVerify/MfaGoogleAuthVerify";
 import { PasswordForm } from "./PasswordForm/PasswordForm";
-
-interface WrapperProps {
-  activationToken: string;
-}
-const AccountCreation404Wrapper: FunctionComponent<WrapperProps> = ({
-  activationToken,
-  children,
-}) => {
-  if (activationToken === undefined) {
-    return <>Loading...</>;
-  }
-  if (activationToken === null) {
-    return <PageNotFound />;
-  }
-  return <>{children}</>;
-};
+import { AccountCreationApi } from "./AccountCreationApiService";
+import { routeFromStatus, UserAccountStatus } from "./UserAccountStatus";
 
 const AccountCreationApp = () => {
-  const dispatch = useDispatch();
-  const activationToken = useSelector<RootState, string>(
-    (state) => state.activationToken
+  // Initialize to loading state on app load
+  const [userAccountStatus, setUserAccountStatus] = useState(
+    UserAccountStatus.LOADING
   );
+  // Used to reroute based on user's status
+  const history = useHistory();
 
+  // Runs once on app load
   useEffect(() => {
-    dispatch(
-      setInitialState({
-        activationToken: getActivationTokenFromUrl(),
-      })
-    );
-  }, [dispatch]);
+    const getStatusAndActivate = async (
+      activationToken: string | null = null
+    ) => {
+      // Ask backend what the user's current account creation status is
+      let status = await AccountCreationApi.getUserStatus(activationToken);
+      // If the user hasn't been activated yet w/ the activation token, do so
+      if (status === UserAccountStatus.PENDING_ACTIVATION && activationToken) {
+        await AccountCreationApi.initialize(activationToken);
+        // Re-retrieve the status since it will have changed after activation
+        status = await AccountCreationApi.getUserStatus();
+      }
+      // Get correct path based on status
+      const newPath = routeFromStatus(status);
+      // Check what current path is
+      const mount = history.location.pathname.split("/uac")[0];
+      const currentPath = history.location.pathname.split("/uac")[1];
+      // If new path is different, reroute
+      if (newPath !== currentPath) {
+        history.push(`${mount}/uac${newPath}`);
+      }
+      // Set the userAccountStatus state, triggering a rerender w/ the Router
+      setUserAccountStatus(status);
+    };
+    const token = getActivationTokenFromUrl();
+    getStatusAndActivate(token);
+  }, [history, userAccountStatus]);
```

Comment:
**question**: Is it intentional here to have `userAccountStatus` in the dependency array? It's not used inside the effect itself but will cause the effect to re-run every time the status changes.

Reply:
Hmmm that's a good question. I suspect so, because it looks like this effect is the only thing that changes `userAccountStatus`. 

I do have a related question - is it the case that a piece of state only needs to be in the dependency array if its value is being read, or should it also be included if the value will be changed (but it's not being read)? My understanding was the latter, but your question seems to imply that it's the former.

Reply:
It only needs the be in the dependency array if it is directly referenced inside the effect callback function. That being said, you may still choose to include it in the dependency array if you want the effect to re-run when that value changes.

Right now, the effect will re-run whenever `userAccountStatus` changes. If you removed `userAccountStatus` from the dependency array, the effect would only run when the `history` object reference changes, which is pretty much only when the component is mounted.

Reply:
IIRC the linter wanted me to include `userAccountStatus` in the dependencies as well, but it was over 2 weeks ago so my memory is fuzzy. Bottom line though is that I really only want this effect to run once on app load.

Reply:
I'm fairly certain you can remove `userAccountStatus` and the linter should be fine. Also, having `history` in the dependency array shouldn't be a big deal, I think it'll just run once (or maybe twice). The history object reference doesn't change! (but might start as undefined?)